BOUNTY.md

```diff
@@ -0,0 +1,118 @@
+
+
+## How to Claim a Bounty
+
+We've created a process to ensure multiple contributors aren’t competing on the same project, and to make sure
+work is properly merged into GraphQL.NET once completed.
+
+1. Read the GitHub issue containing the project's Request For Proposals completely.
+2. If you want to move forward, submit an proposal to the GraphQL.NET team on your suggested implementation by
+   replying to the issue. Be sure to include as many details as requested by the spec, and be sure to address
+   each of the completion criteria. For proposals that include UI changes, mockups are strongly encouraged.
+3. Work with the GraphQL.NET team to finalize plans for the best implementation. Proposals will be judged on their
+   maintainability, design quality, and adherence to spec. Note that other developers may submit competing proposals
+   at this time as well.
+4. If your proposal is accepted, you may begin code implementation that adheres to your proposal.
+5. Submit code as a draft pull request in the appropriate repository so GraphQL.NET community leaders can track progress.
+6. Once code is complete, remove draft status and request a review from the GraphQL.NET maintainers.
+7. Update code with feedback from core GraphQL.NET contributors as needed.
+8. Wait for your code to be merged into the `master` or `develop` branch.
+9. [Submit an expense through Open Collective](https://opencollective.com/graphql-net/expenses/new) for the amount of the bounty.
+
+
+## Collaboration with GraphQL.NET team
+
+Collaboration with the GraphQL.NET team is crucial to having your code pulled into GraphQL.NET. During the proposal process,
+clarify any questions you or your team may have as early as possible. During development, use the pull request commenting
+feature.
+
+
+## Deadlines
+
+If your team does not show code commits or interaction for two weeks at a time as part of an **`in progress`** bounty, then
+the bounty will be released for another team to work on. If the team can demonstrate work on the project, they can reclaim
+the bounty.
+
+If you no longer wish to work on a bounty that has been granted to you, please notify the GraphQL.NET team.
+
+
+## Bounty Valuation
+
+The GraphQL.NET team currently uses a simple rubric to determine how to price a bounty. Generally speaking, RFPs are evaluated
+based on two main criteria:
+
+* How complex is the feature/bugfix?
+    * How long do we expect it may take to complete?
+    * How much specialized knowledge do we expect it will require to implement?
+* How high of a demand is there for this feature/bugfix?
+    * How many people do we expect this to impact?
+    * How frequently are we asked about this feature or bug?
+    * How urgently does this need to be implemented?
+
+In general, issues that are higher in complexity and higher in demand are given greater bounties.
+
+
+## Legal
```

Comment:
@joemcbride Along with the rest of this document, please review this 'Legal' section when you have a minute.

These terms were paraphrased from MS's bounty program here:
https://www.microsoft.com/en-us/msrc/bounty-terms?rtc=1

I selected restrictions from Microsoft's text based on these basis:
1. Age -- due to [COPPA](https://www.ftc.gov/legal-library/browse/rules/childrens-online-privacy-protection-rule-coppa)
2. Individual researcher, etc -- for copyright
3. Resident of countries with US sanctions -- could be a legal requirement
4. MIT license requirements -- for copyright
5. Similarity waiver -- for copyright
6. No guaranteed compensation -- so we can revoke a bounty if needed
7. Submission is own work -- for copyright

I think most of these make sense, but the one regarding US sanctions we may not need.  Specifically, even if there is a US law that applies (which there very likely may not be), technically it is Open Collective that has both received donations and is paying the bounties.  You're simply approving the payment.  (Correct??)  I'd recommend removing that restriction and let Open Collective deal with any issues that may arise.  I really don't want to exclude anyone from this program, and specifically @sungam3r currently falls into this category.

We could also argue that the age restriction falls in the same category.  But as children's privacy is a hot button in almost every country, I'd leave it in there.  I'm sure any 14yo contributors can get their parent to apply for the bounty if need be!

Note that Microsoft has additional restrictions for government employees and Microsoft employees that I didn't feel were applicable here.

Final note: the "submission is own work" restriction could be an issue if someone continues another person's work.  But at the same time, if that was the case, should they be paid a bounty?  I guess I'd just leave it in there for now, and if an issue arises, we can handle it at that time.

Thanks in advance!

Reply:
> I think most of these make sense, but the one regarding US sanctions we may not need. Specifically, even if there is a US law that applies (which there very likely may not be), technically it is Open Collective that has both received donations and is paying the bounties. You're simply approving the payment. (Correct??) I'd recommend removing that restriction and let Open Collective deal with any issues that may arise. I really don't want to exclude anyone from this program, and specifically @sungam3r currently falls into this category.

I agree with this sentiment, that this should be handled at the Open Collective level. Since I'm no lawyer and I'm sure those restrictions change and there are probably a lot of nuances. It would be difficult for the maintainers to try to make that determination.

Reply:
Done.